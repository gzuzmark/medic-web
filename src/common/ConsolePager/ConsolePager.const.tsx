export const PAGE_ITEMS_RANGE = 20;