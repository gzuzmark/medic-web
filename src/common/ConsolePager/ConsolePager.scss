@import "../../assets/styles/01-settings/settings";

$softPageColor: #d8d8d8;
$activePageColor: #707070;

.ConsolePager {
  margin: 25px 0;
  width: 100%;
  .pagination {
    display: flex;
    flex-wrap: wrap;
    list-style-type: none;
    padding-left: 0;
    li {
      a {
        border: 1px solid $softPageColor;
        color: $activePageColor;
        display: block;
        margin-right: -1px;
        padding: 10px;
        text-align: center;
        transition: background 0.3s ease-in-out;
        vertical-align: middle;
        width: 45px;
        &:hover {
          background: #eaeaea;
        }
      }
      &.active {
        a {
          background: $activePageColor;
          border: 1px solid $activePageColor;
          color: white;
        }
      }
      &.disabled {
        a {
          color: $softPageColor;
          &:hover {
            background: white;
          }
        }
      }
    }

  }
}