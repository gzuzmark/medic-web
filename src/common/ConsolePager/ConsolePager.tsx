import * as React from 'react';
import Pagination from "react-js-pagination";
import {PAGE_ITEMS_RANGE} from "./ConsolePager.const";
import './ConsolePager.scss';


interface IPropsConsolePaginator {
    activePage: number;
    pageSize: number
    totalItemsCount: number;
    onChange: (page: number) => void;
    style?: React.CSSProperties;
}

const ConsolePager: React.StatelessComponent<IPropsConsolePaginator> = (props) => {
    if (props.totalItemsCount <= PAGE_ITEMS_RANGE) {
        return null;
    }
    return (
        <div className='ConsolePager' style={{...props.style}}>
            <Pagination
                activePage={props.activePage}
                itemsCountPerPage={props.pageSize}
                totalItemsCount={props.totalItemsCount}
                pageRangeDisplayed={8}
                onChange={props.onChange}
            />
        </div>
    );
};

export default ConsolePager;