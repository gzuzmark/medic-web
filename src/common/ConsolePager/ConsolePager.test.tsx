import { shallow } from 'enzyme';
import 'jest-localstorage-mock';
import * as React from 'react';
import ConsolePager from './ConsolePager';
import {PAGE_ITEMS_RANGE} from "./ConsolePager.const";

describe('ConsolePager Test',() => {
    let props: any;
    let mountedConsolePager: any;
    const getComponent = () => {
        if (!mountedConsolePager) {
            mountedConsolePager = shallow(
                <ConsolePager {...props}>Caja de Mensajes</ConsolePager>
            );
        }
        return mountedConsolePager;
    };

    beforeEach(() => {
        props = {
            activePage: 1,
            onChange: (page: number) => void(0),
            pageSize: 2,
            totalItemsCount: 22
        };
        mountedConsolePager = undefined;
    });

    it("render: render ConsolePager", () => {
        const component = getComponent();
        expect(component.find('.ConsolePager').length).toBe(1);
    });

    it("render: render ConsolePager should be null", () => {
        props = {
            activePage: 1,
            onChange: (page: number) => void(0),
            pageSize: 2,
            totalItemsCount: PAGE_ITEMS_RANGE - 4
        };
        const component = getComponent().find('.ConsolePager');
        expect(component.length).toBe(0);
    });

});
