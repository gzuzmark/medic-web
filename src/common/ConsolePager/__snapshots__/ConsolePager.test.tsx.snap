// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsolePager Test render: render ConsolePager 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConsolePager
    activePage={1}
    onChange={[Function]}
    pageSize={2}
    totalItemsCount={22}
  >
    Caja de Mensajes
  </ConsolePager>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        activeLinkClass={undefined}
        activePage={1}
        firstPageText="«"
        getPageUrl={[Function]}
        hideFirstLastPages={false}
        innerClass="pagination"
        itemClass={undefined}
        itemsCountPerPage={2}
        lastPageText="»"
        linkClass={undefined}
        nextPageText="⟩"
        onChange={[Function]}
        pageRangeDisplayed={8}
        prevPageText="⟨"
        totalItemsCount={22}
      />,
      "className": "ConsolePager",
      "style": Object {},
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeLinkClass": undefined,
        "activePage": 1,
        "firstPageText": "«",
        "getPageUrl": [Function],
        "hideFirstLastPages": false,
        "innerClass": "pagination",
        "itemClass": undefined,
        "itemsCountPerPage": 2,
        "lastPageText": "»",
        "linkClass": undefined,
        "nextPageText": "⟩",
        "onChange": [Function],
        "pageRangeDisplayed": 8,
        "prevPageText": "⟨",
        "totalItemsCount": 22,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          activeLinkClass={undefined}
          activePage={1}
          firstPageText="«"
          getPageUrl={[Function]}
          hideFirstLastPages={false}
          innerClass="pagination"
          itemClass={undefined}
          itemsCountPerPage={2}
          lastPageText="»"
          linkClass={undefined}
          nextPageText="⟩"
          onChange={[Function]}
          pageRangeDisplayed={8}
          prevPageText="⟨"
          totalItemsCount={22}
        />,
        "className": "ConsolePager",
        "style": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeLinkClass": undefined,
          "activePage": 1,
          "firstPageText": "«",
          "getPageUrl": [Function],
          "hideFirstLastPages": false,
          "innerClass": "pagination",
          "itemClass": undefined,
          "itemsCountPerPage": 2,
          "lastPageText": "»",
          "linkClass": undefined,
          "nextPageText": "⟩",
          "onChange": [Function],
          "pageRangeDisplayed": 8,
          "prevPageText": "⟨",
          "totalItemsCount": 22,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
